src/main/java/com/google/firebase/remoteconfig/ParameterGroup.java

```diff
@@ -0,0 +1,138 @@
+    parameters = new HashMap<>();
+  }
+
+  ParameterGroup(@NonNull ParameterGroupResponse parameterGroupResponse) {
+    checkNotNull(parameterGroupResponse);
+    this.parameters = new HashMap<>();
+    if (parameterGroupResponse.getParameters() != null) {
+      for (Map.Entry<String, TemplateResponse.ParameterResponse> entry
+              : parameterGroupResponse.getParameters().entrySet()) {
+        this.parameters.put(entry.getKey(), new Parameter(entry.getValue()));
+      }
+    }
+    this.description = parameterGroupResponse.getDescription();
+  }
+
+  /**
+   * Gets the description of the parameter group.
+   *
+   * @return The description of the parameter or null.
+   */
+  @Nullable
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * Gets the map of parameters that belong to this group.
+   *
+   * @return A non-null map of parameter keys to their optional default values and optional
+   *     conditional values.
+   */
+  @NonNull
+  public Map<String, Parameter> getParameters() {
+    return parameters;
+  }
+
+  /**
+   * Sets the description of the parameter group.
+   * Should not be over 256 characters and may contain any Unicode characters.
+   *
+   * @param description The description of the parameter group.
+   * @return This {@link ParameterGroup}.
+   */
+  public ParameterGroup setDescription(@Nullable String description) {
+    this.description = description;
+    return this;
+  }
+
+  /**
+   * Sets the map of parameters that belong to this group.
+   *
+   * <p>A parameter only appears once per Remote Config template.
+   * An ungrouped parameter appears at the top level, whereas a
+   * parameter organized within a group appears within its group's map of parameters.
+   *
+   * @param parameters A non-null map of parameter keys to their optional default values and
+   *                   optional conditional values.
+   * @return This {@link ParameterGroup} instance.
+   */
+  public ParameterGroup setParameters(
+          @NonNull Map<String, Parameter> parameters) {
+    checkNotNull(parameters, "parameters must not be null.");
+    this.parameters = parameters;
+    return this;
+  }
+
+  ParameterGroupResponse toParameterGroupResponse() {
+    Map<String, ParameterResponse> parameterResponses = new HashMap<>();
+    for (Map.Entry<String, Parameter> entry : this.parameters.entrySet()) {
+      parameterResponses.put(entry.getKey(), entry.getValue().toParameterResponse());
+    }
+    return new ParameterGroupResponse()
+            .setDescription(this.description)
+            .setParameters(parameterResponses);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    ParameterGroup that = (ParameterGroup) o;
+    return Objects.equals(description, that.description)
+            && Objects.equals(parameters, that.parameters);
```

Comment:
Does this work on collections like maps?

Reply:
Yes, it seems to be working fine on maps. `Objects.equals(o1, o2)` will end up calling `o1.equals(o2)` and that should work fine since we have implemented `equals` in the custom types `Parameter` etc.
https://docs.oracle.com/javase/7/docs/api/java/util/Map.html#equals(java.lang.Object)

I have added test cases for equality in `ParameterGroupTest.java`. I can change that test to use two separate collections to make it more explicit if that helps...